relayer/docker.go

```diff
@@ -257,27 +263,56 @@ func (r *DockerRelayer) GetChannels(ctx context.Context, rep ibc.RelayerExecRepo
 	ctx, cancel := context.WithTimeout(ctx, 3*time.Minute)
 	defer cancel()
 
-	exitCode, stdout, stderr, err := r.NodeJob(ctx, rep, cmd)
+	stdout, stderr, err := r.Exec(ctx, rep, cmd, nil)
 	if err != nil {
-		return nil, dockerutil.HandleNodeJobError(exitCode, stdout, stderr, err)
+		return nil, err
 	}
 
-	return r.c.ParseGetChannelsOutput(stdout, stderr)
+	return r.c.ParseGetChannelsOutput(string(stdout), string(stderr))
 }
 
 func (r *DockerRelayer) GetConnections(ctx context.Context, rep ibc.RelayerExecReporter, chainID string) (ibc.ConnectionOutputs, error) {
 	cmd := r.c.GetConnections(chainID, r.NodeHome())
-	exitCode, stdout, stderr, err := r.NodeJob(ctx, rep, cmd)
+	stdout, stderr, err := r.Exec(ctx, rep, cmd, nil)
 	if err != nil {
-		return nil, dockerutil.HandleNodeJobError(exitCode, stdout, stderr, err)
+		return nil, err
 	}
 
-	return r.c.ParseGetConnectionsOutput(stdout, stderr)
+	return r.c.ParseGetConnectionsOutput(string(stdout), string(stderr))
 }
 
 func (r *DockerRelayer) LinkPath(ctx context.Context, rep ibc.RelayerExecReporter, pathName string, opts ibc.CreateChannelOptions) error {
 	cmd := r.c.LinkPath(pathName, r.NodeHome(), opts)
-	return dockerutil.HandleNodeJobError(r.NodeJob(ctx, rep, cmd))
+	_, _, err := r.Exec(ctx, rep, cmd, nil)
+	return err
+}
+
+func (r *DockerRelayer) Exec(ctx context.Context, rep ibc.RelayerExecReporter, cmd []string, env []string) (stdout, stderr []byte, err error) {
+	job := dockerutil.NewImage(r.log, r.client, r.networkID, r.testName, r.containerImage().Repository, r.containerImage().Version)
+	opts := dockerutil.ContainerOptions{
+		Env:   env,
+		Binds: r.Bind(),
+	}
+
+	var exitCode int
+	startedAt := time.Now()
+	stdout, stderr, err = job.Run(ctx, cmd, opts)
+	if err != nil {
+		exitCode = 1 // TODO this is possibly not correct, need some feedback on this
```

Comment:
I'm not 100% sure this is achieving exactly the same behavior as we had before.

Reply:
I'll get a follow up PR opened separately, tomorrow at some point, that brings in the `ContainerError` type!

Reply:
Capturing non-zero exits as an error is subtle. This is much like writing an HTTP client library that treats a completed request with a 404 status as an error. 

There is a difference between "I ran the command to execution, and it exited 1" versus "I failed to run the command, because the binary `rly` was not on the path".

It looks like the `job.Run` signature is insufficient to distinguish this detail right now.